### PR TITLE
default limit 100 queries

### LINKED ISSUE #597: Add SQL query `limit` parameter to the search functions
Each language should already have a `Search(string sqlQuery)` function. Each language should also already have a `Search()` function with a sane SQL query default. This ticket is a request to add the `limit` parameter as an optional argument to the sane defaults `Search()` function so that the user can limit the count of results returned. Default remains to return all of them.

### Details of `limit` parameter
* Essentially just switches from `SELECT * FROM c` to `SELECT * FROM c LIMIT [N=10]` (where `N` can be changed, obviously)
* Do we want a default `order by` clause as well to ensure a stable query?

### Requirements
1. Hide `continuation_token` and `has_more` parameters, return everything from the user's query
2. Optional limit parameter on the sane-defaults query

### WalletService
Languages:
- [x] .NET
- [x] Go
- [x] Java
- [x] Node
- [x] Python
- [x] Ruby
- [x] Swift
- [x] Web

### TrustRegistryService
Languages:
- [x] .NET
- [x] Go
- [x] Java
- [x] Node
- [x] Python
- [x] Ruby
- [x] Swift
- [x] Web

### PR DESCRIPTION
**Pull request by fundthmcalculus**
closes #597

### CODE REVIEW

File: docs/reference/services/template-service.md [hunk 1]
```diff
@@ -119,7 +119,7 @@ Template searching works very similarly to Wallet searching. Please refer to [Wa
 
 ### Basic Search
 
-The default query used in the commands below returns a full wallet result set. The query is `SELECT * FROM c`.
+The default query used in the commands below returns a first-100 wallet result set. The query is `SELECT * FROM c LIMIT 100`.
```
Comment (geel9) [inline docs/reference/services/template-service.md]:
```suggestion
The default query used in the commands below returns the first 100 items in the template result set. The query is `SELECT * FROM c LIMIT 100`.
```

File: docs/reference/services/wallet-service.md [hunk 2]
```diff
@@ -77,7 +77,7 @@ This approach allows us to give developers full control over how data is retriev
 
 ### Basic Search
 
-The default query used in the commands below returns a full wallet result set. The query is `SELECT * FROM c`.
+The default query used in the commands below returns a first-100 wallet result set. The query is `SELECT * FROM c LIMIT 100`.
```
Comment (geel9) [inline docs/reference/services/wallet-service.md]:
```suggestion
The default query used in the commands below returns the first 100 items in the wallet result set. The query is `SELECT * FROM c LIMIT 100`.
```

**Review by geel9 — APPROVED** [review]

File: cli/src/cli.yaml [hunk 3]
```diff
@@ -95,7 +95,7 @@ subcommands:
                   value_name: SQL_QUERY
                   help: >-
                     (Optional) Search query to execute (Default is "SELECT *
-                    FROM c"). To see all available query options, please check
+                    FROM c  LIMIT 100"). To see all available query options, please check
```
Comment (fundthmcalculus) [inline cli/src/cli.yaml]:
```suggestion
                    FROM c OFFSET 0 LIMIT 100"). To see all available query options, please check
```

File: cli/src/cli.yaml [hunk 4]
```diff
@@ -305,7 +305,7 @@ subcommands:
                   value_name: SQL query
                   help: |
                     The SQL query to search the registry.
-                    Default value is "SELECT c.data, c.id, c.type FROM c".
+                    Default value is "SELECT c.data, c.id, c.type FROM c LIMIT 100".
```
Comment (fundthmcalculus) [inline cli/src/cli.yaml]:
```suggestion
                    Default value is "SELECT c.data, c.id, c.type FROM c OFFSET 0 LIMIT 100".
```

File: cli/src/parser/template.rs [hunk 5]
```diff
@@ -156,7 +156,7 @@ fn list(args: &ArgMatches) -> Result<TemplateCommand, Error> {
     Ok(TemplateCommand::List(ListTemplatesArgs {
         query: args
             .value_of("query")
-            .map_or("SELECT * FROM c".to_string(), |x| x.to_string()),
+            .map_or("SELECT * FROM c LIMIT 100".to_string(), |x| x.to_string()),
```
Comment (fundthmcalculus) [inline cli/src/parser/template.rs]:
```suggestion
            .map_or("SELECT * FROM c OFFSET 0 LIMIT 100".to_string(), |x| x.to_string()),
```

File: cli/src/parser/template.rs [hunk 6]
```diff
@@ -167,7 +167,7 @@ fn search(args: &ArgMatches) -> Result<TemplateCommand, Error> {
     Ok(TemplateCommand::Search(SearchTemplatesArgs {
         query: args
             .value_of("query")
-            .map_or("SELECT * FROM c".to_string(), |x| x.to_string()),
+            .map_or("SELECT * FROM c LIMIT 100".to_string(), |x| x.to_string()),
```
Comment (fundthmcalculus) [inline cli/src/parser/template.rs]:
```suggestion
            .map_or("SELECT * FROM c OFFSET 0 LIMIT 100".to_string(), |x| x.to_string()),
```

File: cli/src/services/trustregistry.rs [hunk 7]
```diff
@@ -27,7 +27,7 @@ async fn search(args: &SearchArgs, config: &CliConfig) -> Result<Output, Error>
     let query = args
         .query
         .as_ref()
-        .map_or("SELECT * FROM c".to_string(), |q| q.to_owned());
+        .map_or("SELECT * FROM c LIMIT 100".to_string(), |q| q.to_owned());
```
Comment (fundthmcalculus) [inline cli/src/services/trustregistry.rs]:
```suggestion
        .map_or("SELECT * FROM c OFFSET 0 LIMIT 100".to_string(), |q| q.to_owned());
```

File: cli/src/services/wallet.rs [hunk 8]
```diff
@@ -33,7 +33,7 @@ pub(crate) fn execute(args: &Command, config: CliConfig) -> Result<Output, Error
 async fn search(args: &SearchArgs, config: CliConfig) -> Result<Output, Error> {
     let query = args
         .query
-        .map_or("SELECT c.data, c.id, c.type FROM c".to_string(), |q| {
+        .map_or("SELECT c.data, c.id, c.type FROM c LIMIT 100".to_string(), |q| {
```
Comment (fundthmcalculus) [inline cli/src/services/wallet.rs]:
```suggestion
        .map_or("SELECT c.data, c.id, c.type FROM c OFFSET 0 LIMIT 100".to_string(), |q| {
```

File: docs/reference/services/template-service.md [hunk 9]
```diff
@@ -119,7 +119,7 @@ Template searching works very similarly to Wallet searching. Please refer to [Wa
 
 ### Basic Search
 
-The default query used in the commands below returns a full wallet result set. The query is `SELECT * FROM c`.
+The default query used in the commands below returns the first 100 items in the template result set. The query is `SELECT * FROM c LIMIT 100`.
```
Comment (fundthmcalculus) [inline docs/reference/services/template-service.md]:
```suggestion
The default query used in the commands below returns the first 100 items in the template result set. The query is `SELECT * FROM c OFFSET 0 LIMIT 100`.
```

File: docs/reference/services/wallet-service.md [hunk 10]
```diff
@@ -77,7 +77,7 @@ This approach allows us to give developers full control over how data is retriev
 
 ### Basic Search
 
-The default query used in the commands below returns a full wallet result set. The query is `SELECT * FROM c`.
+The default query used in the commands below returns the first 100 items in the wallet result set. The query is `SELECT * FROM c LIMIT 100`.
```
Comment (fundthmcalculus) [inline docs/reference/services/wallet-service.md]:
```suggestion
The default query used in the commands below returns the first 100 items in the wallet result set. The query is `SELECT * FROM c OFFSET 0 LIMIT 100`.
```

File: dotnet/Trinsic/TrustRegistryService.cs [hunk 11]
```diff
@@ -129,15 +129,15 @@ internal TrustRegistryService(ITokenProvider tokenProvider, IOptions<ServiceOpti
     /// <returns></returns>
     public async Task<SearchRegistryResponse> SearchRegistryAsync(SearchRegistryRequest request) {
         if (String.IsNullOrWhiteSpace(request.Query))
-            request.Query = "SELECT * FROM c";
+            request.Query = "SELECT * FROM c LIMIT 100";
```
Comment (fundthmcalculus) [inline dotnet/Trinsic/TrustRegistryService.cs]:
```suggestion
            request.Query = "SELECT * FROM c OFFSET 0 LIMIT 100";
```

File: dotnet/Trinsic/TrustRegistryService.cs [hunk 12]
```diff
@@ -129,15 +129,15 @@ internal TrustRegistryService(ITokenProvider tokenProvider, IOptions<ServiceOpti
...
         
         var response = await Client.SearchRegistryAsync(request, await BuildMetadataAsync(request));
         return response;
     }
 
     public SearchRegistryResponse SearchRegistry(SearchRegistryRequest request) {
         if (String.IsNullOrWhiteSpace(request.Query))
-            request.Query = "SELECT * FROM c";
+            request.Query = "SELECT * FROM c LIMIT 100";
```
Comment (fundthmcalculus) [inline dotnet/Trinsic/TrustRegistryService.cs]:
```suggestion
            request.Query = "SELECT * FROM c OFFSET 0 LIMIT 100";
```

File: dotnet/Trinsic/WalletService.cs [hunk 13]
```diff
@@ -36,7 +36,7 @@ internal WalletService(ITokenProvider tokenProvider, IOptions<ServiceOptions> op
     /// <returns></returns>
     public async Task<SearchResponse> SearchAsync(SearchRequest request) {
         if (string.IsNullOrWhiteSpace(request.Query))
-            request.Query = "SELECT c.id, c.type, c.data FROM c";
+            request.Query = "SELECT c.id, c.type, c.data FROM c LIMIT 100";
```
Comment (fundthmcalculus) [inline dotnet/Trinsic/WalletService.cs]:
```suggestion
            request.Query = "SELECT c.id, c.type, c.data FROM c OFFSET 0 LIMIT 100";
```

File: dotnet/Trinsic/WalletService.cs [hunk 14]
```diff
@@ -51,7 +51,7 @@ internal WalletService(ITokenProvider tokenProvider, IOptions<ServiceOptions> op
     /// <returns></returns>
     public SearchResponse Search(SearchRequest request) {
         if (string.IsNullOrWhiteSpace(request.Query))
-            request.Query = "SELECT c.id, c.type, c.data FROM c";
+            request.Query = "SELECT c.id, c.type, c.data FROM c LIMIT 100";
```
Comment (fundthmcalculus) [inline dotnet/Trinsic/WalletService.cs]:
```suggestion
            request.Query = "SELECT c.id, c.type, c.data FROM c OFFSET 0 LIMIT 100";
```

File: web/src/WalletService.ts [hunk 15]
```diff
@@ -26,7 +26,7 @@ export class WalletService extends ServiceBase {
 
   public async search(
     request: SearchRequest = SearchRequest.fromPartial({
-      query: "SELECT * FROM c",
+      query: "SELECT c.id, c.type, c.data FROM c LIMIT 100",
```
Comment (fundthmcalculus) [inline web/src/WalletService.ts]:
```suggestion
      query: "SELECT c.id, c.type, c.data FROM c OFFSET 0 LIMIT 100",
```

File: web/src/TrustRegistryService.ts [hunk 16]
```diff
@@ -98,7 +98,7 @@ export class TrustRegistryService extends ServiceBase {
   }
 
   public async searchRegistry(
-    request: SearchRegistryRequest = SearchRegistryRequest.fromPartial({query: "SELECT * FROM c"})
+    request: SearchRegistryRequest = SearchRegistryRequest.fromPartial({query: "SELECT * FROM c LIMIT 100"})
```
Comment (fundthmcalculus) [inline web/src/TrustRegistryService.ts]:
```suggestion
    request: SearchRegistryRequest = SearchRegistryRequest.fromPartial({query: "SELECT * FROM c OFFSET 0 LIMIT 100"})
```

File: ruby/lib/services/wallet_service.rb [hunk 17]
```diff
@@ -16,7 +16,7 @@ def initialize(service_options = nil)
     def search(request = nil)
       # request = Wallet_V1::SearchRequest.new(query: query)
       request = request || Wallet_V1::SearchRequest.new
-      request.query = request.query.empty? ? "SELECT c.id, c.type, c.data FROM c" : request.query
+      request.query = request.query.empty? ? "SELECT c.id, c.type, c.data FROM c LIMIT 100" : request.query
```
Comment (fundthmcalculus) [inline ruby/lib/services/wallet_service.rb]:
```suggestion
      request.query = request.query.empty? ? "SELECT c.id, c.type, c.data FROM c OFFSET 0 LIMIT 100" : request.query
```

File: ruby/lib/services/trust_registry_service.rb [hunk 18]
```diff
@@ -54,7 +54,7 @@ def check_verifier_status(request)
     def search_registry(request = nil)
       # request = TrustRegistry_V1::SearchRegistryRequest.new(query: query)
       request ||= TrustRegistry_V1::SearchRegistryRequest.new
-      request.query = request.query.empty? ? "SELECT * FROM c" : request.query
+      request.query = request.query.empty? ? "SELECT * FROM c LIMIT 100" : request.query
```
Comment (fundthmcalculus) [inline ruby/lib/services/trust_registry_service.rb]:
```suggestion
      request.query = request.query.empty? ? "SELECT * FROM c OFFSET 0 LIMIT 100" : request.query
```

File: python/trinsic/wallet_service.py [hunk 19]
```diff
@@ -36,7 +36,7 @@ async def search(self, *, request: SearchRequest = None) -> SearchResponse:
             The search response object information
         """
         request = request or SearchRequest()
-        request.query = request.query or "SELECT c.id, c.type, c.data FROM c"
+        request.query = request.query or "SELECT c.id, c.type, c.data FROM c LIMIT 100"
```
Comment (fundthmcalculus) [inline python/trinsic/wallet_service.py]:
```suggestion
        request.query = request.query or "SELECT c.id, c.type, c.data FROM c OFFSET 0 LIMIT 100"
```

File: python/trinsic/trustregistry_service.py [hunk 20]
```diff
@@ -148,7 +148,7 @@ async def search_registry(
             [SearchRegistryResponse](/reference/proto/#searchregistryresponse)
         """
         request = request or SearchRegistryRequest()
-        request.query = request.query or "SELECT * FROM c"
+        request.query = request.query or "SELECT * FROM c LIMIT 100"
```
Comment (fundthmcalculus) [inline python/trinsic/trustregistry_service.py]:
```suggestion
        request.query = request.query or "SELECT * FROM c OFFSET 0 LIMIT 100"
```

File: python/trinsic/credentialtemplates_service.py [hunk 21]
```diff
@@ -51,7 +51,7 @@ async def list(
     async def search(
         self, *, request: SearchCredentialTemplatesRequest
     ) -> SearchCredentialTemplatesResponse:
-        request.query = request.query or "SELECT * from c"
+        request.query = request.query or "SELECT * from c LIMIT 100"
```
Comment (fundthmcalculus) [inline python/trinsic/credentialtemplates_service.py]:
```suggestion
        request.query = request.query or "SELECT * from c OFFSET 0 LIMIT 100"
```

File: java/src/main/java/trinsic/services/WalletServiceKt.kt [hunk 22]
```diff
@@ -14,7 +14,7 @@ class WalletServiceKt(
     @Throws(InvalidProtocolBufferException::class, DidException::class)
     suspend fun search(request: SearchRequest): SearchResponse {
         var request = request
-        if (request.query.isBlank()) request = SearchRequest.newBuilder(request).setQuery("SELECT c.id, c.type, c.data FROM c").build()
+        if (request.query.isBlank()) request = SearchRequest.newBuilder(request).setQuery("SELECT c.id, c.type, c.data FROM c LIMIT 100").build()
```
Comment (fundthmcalculus) [inline java/src/main/java/trinsic/services/WalletServiceKt.kt]:
```suggestion
        if (request.query.isBlank()) request = SearchRequest.newBuilder(request).setQuery("SELECT c.id, c.type, c.data FROM c OFFSET 0 LIMIT 100").build()
```

File: java/src/main/java/trinsic/services/WalletService.java [hunk 23]
```diff
@@ -18,7 +18,7 @@ public WalletService(Options.ServiceOptions options) {
     }
 
     public ListenableFuture<UniversalWalletOuterClass.SearchResponse> search(UniversalWalletOuterClass.SearchRequest request) throws InvalidProtocolBufferException, DidException {
-        if (request.getQuery().isBlank()) request = UniversalWalletOuterClass.SearchRequest.newBuilder(request).setQuery("SELECT c.id, c.type, c.data FROM c").build();
+        if (request.getQuery().isBlank()) request = UniversalWalletOuterClass.SearchRequest.newBuilder(request).setQuery("SELECT c.id, c.type, c.data FROM c LIMIT 100").build();
```
Comment (fundthmcalculus) [inline java/src/main/java/trinsic/services/WalletService.java]:
```suggestion
        if (request.getQuery().isBlank()) request = UniversalWalletOuterClass.SearchRequest.newBuilder(request).setQuery("SELECT c.id, c.type, c.data FROM c OFFSET 0 LIMIT 100").build();
```

File: java/src/main/java/trinsic/services/TrustRegistryServiceKt.kt [hunk 24]
```diff
@@ -94,7 +94,7 @@ class TrustRegistryServiceKt(
     suspend fun searchRegistry(request: SearchRegistryRequest): SearchRegistryResponse {
         var request = request
         if (request.query.isBlank()) request =
-            SearchRegistryRequest.newBuilder(request).setQuery("SELECT * FROM c").build()
+            SearchRegistryRequest.newBuilder(request).setQuery("SELECT * FROM c LIMIT 100").build()
```
Comment (fundthmcalculus) [inline java/src/main/java/trinsic/services/TrustRegistryServiceKt.kt]:
```suggestion
            SearchRegistryRequest.newBuilder(request).setQuery("SELECT * FROM c OFFSET 0 LIMIT 100").build()
```

File: java/src/main/java/trinsic/services/TrustRegistryService.java [hunk 25]
```diff
@@ -67,7 +67,7 @@ public ListenableFuture<TrustRegistryOuterClass.SearchRegistryResponse> searchRe
     }
 
     public ListenableFuture<TrustRegistryOuterClass.SearchRegistryResponse> searchRegistry(TrustRegistryOuterClass.SearchRegistryRequest request) throws InvalidProtocolBufferException, DidException {
-        if (request.getQuery().isBlank()) request = TrustRegistryOuterClass.SearchRegistryRequest.newBuilder(request).setQuery("SELECT * FROM c").build();
+        if (request.getQuery().isBlank()) request = TrustRegistryOuterClass.SearchRegistryRequest.newBuilder(request).setQuery("SELECT * FROM c LIMIT 100").build();
```
Comment (fundthmcalculus) [inline java/src/main/java/trinsic/services/TrustRegistryService.java]:
```suggestion
        if (request.getQuery().isBlank()) request = TrustRegistryOuterClass.SearchRegistryRequest.newBuilder(request).setQuery("SELECT * FROM c OFFSET 0 LIMIT 100").build();
```

File: go/services/wallet_service.go [hunk 26]
```diff
@@ -40,7 +40,7 @@ func (w *walletBase) Search(userContext context.Context, request *sdk.SearchRequ
 	}
 
 	if len(request.Query) == 0 {
-		request.Query = "SELECT c.id, c.type, c.data FROM c"
+		request.Query = "SELECT c.id, c.type, c.data FROM c LIMIT 100"
```
Comment (fundthmcalculus) [inline go/services/wallet_service.go]:
```suggestion
		request.Query = "SELECT c.id, c.type, c.data FROM c OFFSET 0 LIMIT 100"
```

File: go/services/trustregistry_service.go [hunk 27]
```diff
@@ -148,7 +148,7 @@ func (t *trustRegistryBase) SearchRegistry(userContext context.Context, request
 	}
 
 	if request.Query == "" {
-		request.Query = "SELECT * FROM c"
+		request.Query = "SELECT * FROM c LIMIT 100"
```
Comment (fundthmcalculus) [inline go/services/trustregistry_service.go]:
```suggestion
		request.Query = "SELECT * FROM c OFFSET 0 LIMIT 100"
```

**Review by fundthmcalculus — COMMENTED** [review]
good catch @geel9

**Pull request merged**